base/setup/reactos/drivepage.c

```diff
@@ -78,15 +304,54 @@ MoreOptDlgProc(HWND hwndDlg,
             {
                 case IDOK:
                 {
-                    GetDlgItemTextW(hwndDlg, IDC_PATH,
-                                    pSetupData->USetupData.InstallationDirectory,
-                                    ARRAYSIZE(pSetupData->USetupData.InstallationDirectory));
-                    EndDialog(hwndDlg, IDOK);
+                    HWND hEdit;
+                    BOOL bIsValid;
+                    WCHAR InstallDir[MAX_PATH];
+                    INT iItem;
+                    UINT uBldrLoc = CB_ERR;
+
+                    /*
+                     * Retrieve the installation path and verify its validity.
+                     * Check for the validity of the installation directory and
+                     * pop up an error if this is not the case.
+                     */
+                    hEdit = GetDlgItem(hDlg, IDC_PATH);
+                    bIsValid = (GetWindowTextLengthW(hEdit) < _countof(InstallDir)); // && IsValidInstallDirectory(InstallDir);
+                    GetWindowTextW(hEdit, InstallDir, _countof(InstallDir));
+                    bIsValid = bIsValid && IsValidInstallDirectory(InstallDir);
+
+                    if (!bIsValid)
+                    {
+                        // ERROR_DIRECTORY_NAME
+                        DisplayMessage(hDlg,
+                                       MB_OK | MB_ICONERROR,
+                                       L"Invalid installation directory",
+                                       L"The ReactOS installation directory must be composed of valid 8.3 components, "
```

Comment:
```suggestion
                                       L"The ReactOS installation directory path must follow 8.3 naming scheme, "
```

Reply:
Just to clarify what I meant in the original sentence, I was kind of emphasizing that each path component (what's between path separators) must be in 8.3 format. (Example of allowed path: `\ReactOS\new\0_4.14`)

Reply:
I will update my new translations once we've settle this.

Reply:
Addressed, see latest commit (en-US.rc file)